c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -316,25 +353,21 @@ struct THCCachingAllocator
     allocated_blocks.erase(it);
     block->allocated = false;
 
-    get_stats_for_device(block->device).decreaseAllocated(block->size);
+    DeviceStats& stats = get_stats_for_device(block->device);
+    StatTypes stat_types;
+    stat_types[static_cast<size_t>(StatType::AGGREGATE)] = true;
+    stat_types[static_cast<size_t>(get_stat_type_for_pool(*(block->pool)))] = true;
```

Comment:
nit: To be honest I'm not sure why there's a bitset here at all. You *always* update the aggregate amounts at all call sites. So honestly `update_stat_array` should just take the pool type (and then handling upgrading aggregate stats internally.)

Reply:
I decided to have the vector implementation (now bitset) to support future cases of tracking with non-disjoint categories. I feel it's not super high-cost in either perf or readability to keep it in, but I can change it if you still feel strongly that it should be simplified.